pyomo/contrib/parmest/parmest.py

```diff
@@ -578,10 +688,60 @@ def _Q_opt(
                 the constant cancels out. (was scaled by 1/n because it computes an
                 expected value.)
                 '''
-                cov = 2 * sse / (n - l) * inv_red_hes
-                cov = pd.DataFrame(
-                    cov, index=thetavals.keys(), columns=thetavals.keys()
-                )
+                if self.obj_function == 'SSE':
```

Comment:
Suggestion: move this to a separate function. Also, aren't compute_FIM and compute_cov redundant, as they are inverses of each other?